dali/pipeline/operator/builtin/input_operator.cc

```diff
@@ -115,4 +115,60 @@ InputOperator<GPUBackend>::ForwardCurrentData(TensorList<GPUBackend> &target, cu
   }
 }
 
+
+template<>
+void InputOperator<MixedBackend>::ForwardCurrentData(
+        TensorList<GPUBackend> &target, cudaStream_t stream) {
+  std::list<uptr_tl_type> tensor_list_elm;
+  InputSourceState state_info;
+  {
+    std::unique_lock<std::mutex> busy_lock(busy_m_);
+    tensor_list_elm = tl_data_.PopFront();
+    state_info = state_.front();
+    state_.pop_front();
+  }
+  target.Copy(*tensor_list_elm.front(), stream);
+
+  tensor_list_elm.front()->set_order(internal_copy_order_);
+
+  RecycleBuffer(tensor_list_elm);
+}
+
+
+template<>
+void InputOperator<MixedBackend>::ForwardCurrentData(
+        TensorList<CPUBackend> &target, ThreadPool &thread_pool) {
+  std::list<uptr_tl_type> tensor_list_elm;
+  {
+    std::unique_lock<std::mutex> busy_lock(busy_m_);
+    tensor_list_elm = tl_data_.PopFront();
+    state_.pop_front();
+  }
+  // if the output is pinned and input not it needs to be copied
+  if (target.is_pinned() && !tensor_list_elm.front()->is_pinned()) {
+    const auto &shapes = tensor_list_elm.front()->shape();
+    auto curr_batch_size = shapes.num_samples();
+    target.Resize(shapes, tensor_list_elm.front()->type());
+
+    // as we copy element by element and the output is contiguous we need to set layout
+    // for the whole output not each element(view)
+    target.SetLayout(tensor_list_elm.front()->GetLayout());
+
+    for (int sample_id = 0; sample_id < curr_batch_size; ++sample_id) {
+      thread_pool.AddWork(
+              [&target, sample_id, &tensor_list_elm](int tid) {
+                  target.CopySample(sample_id, *tensor_list_elm.front(), sample_id,
+                                    AccessOrder::host());
```

Comment:
```suggestion
                  target.CopySample(sample_id, *tensor_list_elm.front(), sample_id);
```
leave the order to be figured out from the arguments?